Indicators/IndicatorExtensions.cs

```diff
@@ -70,7 +70,7 @@ public static T Of<T>(this T second, IIndicator first, bool waitForFirstToReady
         /// <param name="weight">Indicator that provides the average weights</param>
         /// <param name="period">Average period</param>
         /// <returns>Indicator that results of the average of first by weights given by second</returns>
-        public static CompositeIndicator WeightedBy<T, TWeight>(this IndicatorBase<T> value, TWeight weight, int period)
+        public static ResetCompositeIndicator WeightedBy<T, TWeight>(this IndicatorBase<T> value, TWeight weight, int period)
```

Comment:
`ResetCompositeIndicator` can be a `CompositeIndicator` that overrides the `Reset` method, that way we don't change the `WeightedBy` API (keep returning `CompositeIndicator`) and users can still access every member of the `CompositeIndicator` as per usual.